### ISSUE TITLE
table alias in keyword list not correctly parsed

### ISSUE THREAD

**Issue by niyue** [macbre/sql-metadata#277]
I found a query with table alias that cannot be correctly parsed:
```SQL
SELECT
    cmp.id
FROM
    company cmp
```
* `parser.tables_aliases` is expected to return `{"cmp": "company"}`, however, it returns `{}`.

This is a parsing issue I found in a large query, and I narrowed down the problem to the table alias name `cmp` used here.

It looks like `cmp` is a keyword in Hive defined in sqlparse (https://github.com/andialbrecht/sqlparse/blob/e66046785b816e5c2d22f101f36faefd19c4a771/sqlparse/keywords.py), which causes this issue. But the query parsed in my case is for postgres, which makes this result not expected (it is a valid query for postgres). 
I think it is likely all the keywords defined in `keywords.py` cannot be used as table aliases correctly, which is a huge list and people may run into such an issue without realizing it.

Update:
I can confirm all the keywords in the `keywords.py` cannot be correctly parsed as table alias. For example, I verified two more cases:
```SQL
SELECT
   system.id
FROM
    my_table system
```

```SQL
SELECT
   result.id
FROM
    my_table result
```

Neither of the above two works.
This keyword list is non trivial (nearly 1000 words), some of them are pretty common words, to name a few, `SYSTEM` `RESULT` `ZONE` `TOAST` `TEXT` `ARCHIVE` `ARCHIVELOG` `BACKUP` `BLOCK` `BODY` `ELEMENTS` `EXCHANGE` `FTP` `LEN` `SIGN` `REDUCE` `STACK` `SUMMARY` `VIEWS`, so people may run into this issue accidentally.

**Comment by collerek** [comment]
You can do thia and this should work but you need to quote the name in that case so i.e.

```sql
SELECT
   system.id
FROM
    my_table `system`
```

Dividing `sql-metadata` into dialects sounds interesting but it would be a huge complication so for now it's outside of the scope.

**Comment by niyue** [comment]
>  you need to quote the name in that case
The simplest fix is actually rename the table alias to non keywords, for example:
```
SELECT
   sys.id
FROM
    my_table sys
```

But I think if a query is a valid query, it is parser's responsibility to parse it correctly instead of asking SQL author to rewrite their queries, and sometimes this is not practically feasible. 

In my case (and probably in other users's cases as well), we use sql-metadata for parsing SQL in application logs. The application development team and the SQL analysis team are two different teams, so it is not possible to ask application development team to re-write their queries just for the case of analysis. 

I think in most cases, users will only validate their query against their database's SQL parser, if the database's SQL parser says this is valid query and can execute it, then it is a valid query, and the third party SQL parser should try to be compatible.


In this case:
1) `SELECT system.id FROM my_table system` is a valid query for MySQL/Postgres/SQLite/MS SQL/Apache Hive since `system` is not a keyword in these databases. And these databases are claimed to be supported in the project's description. It is probably not a valid query for Oracle since `system` is a keyword for Oracle. But Oracle is not claimed to be supported in the project's description. 
2) Since it is a valid query for these database, it is not reasonable and sometimes not practical to ask users to re-write their query (why would a Postgres db user need to re-write the query because Oracle introduces a new keyword?)
3) the keyword list is non trivial (~1000 keywords  and the list will keep growing), there is no way all the users will remember all of them when authoring SQL queries.

**Comment by niyue** [comment]
> Dividing sql-metadata into dialects

Probably I don't make it clearly enough. I don't propose dividing `sql-metadata` into dialects, and the PR doesn't have any database specific change, it is a general change for all databases, and simply allowing using the SQL keywords as table alias , for all databases.

**Comment by collerek** [comment]
closed by #278
